### PR TITLE
Beta14/Mithril 2

### PR DESCRIPTION
**Pull request by karaok491**

### CODE REVIEW

File: js/src/admin/components/EditPageModal.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+import withAttr from 'flarum/utils/withAttr';
 
 /**
  * The `EditPageModal` component shows a modal dialog which allows the user
  * to create or edit a page.
  */
 export default class EditPageModal extends Modal {
-    init() {
-        super.init();
+    oninit(vdom) {
```
Comment (askvortsov1) [inline js/src/admin/components/EditPageModal.js]: vnode, not vdom

File: js/src/admin/components/EditPageModal.js [hunk 2]
```diff
@@ -136,7 +137,7 @@ export default class EditPageModal extends Modal {
             .then(this.hide.bind(this))
             .catch(() => {
                 this.loading = false;
-                m.redraw();
+                m.redraw.sync();
```
Comment (askvortsov1) [inline js/src/admin/components/EditPageModal.js]: Why are these synchronous now?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
The parts that have been changed look good to me, with a few issues:

- vnode should be used instead of vdom for `oninit`; no difference in functionality, but better to follor convention
- I'm seeing a lot of synchronous redraws, is this on purpose?

File: js/src/admin/components/EditPageModal.js (hunk 2 above)
Comment (karaok491) [reply js/src/admin/components/EditPageModal.js]: My bad, mis-read the docs

File: js/src/admin/components/EditPageModal.js (hunk 1 above)
Comment (karaok491) [reply js/src/admin/components/EditPageModal.js]: Doh!

**Pull request merged**